### PR TITLE
feat(onyx-736): view arworks screen for alert matches

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-736]

### Description

Implement "View Artworks" screen for alert matches.

Small demo:

<video src="https://github.com/artsy/eigen/assets/3934579/58a43c67-e9e3-43fc-a38c-b69dd4849c1d" width="400px;" />

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Implement "View Artworks" screen for alert matches - @nickskalkin 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-736]: https://artsyproduct.atlassian.net/browse/ONYX-736?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Implement "View Artworks" screen for alert matches - @nickskalkin)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 69707878fb834cebb4ab47f0daf039c6646e02f5
</p>

File: src/app/Scenes/SavedSearchAlert/AlertArtworksPills.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+export const AlertArtworksPills: FC<AlertArtworksPillsProps> = ({ alertId }) => {
+  const data = useLazyLoadQuery<AlertArtworksPillsQuery>(alertArtworksPillsQuery, {
+    alertId: alertId,
+  })
+
+  const pills = data.me?.alert?.pills || []
+
+  return (
+    <Flex flexDirection="row" flexWrap="wrap">
+      {pills.map((pill: any) => {
```
Comment (olerichter00) [inline src/app/Scenes/SavedSearchAlert/AlertArtworksPills.tsx:20]:
```suggestion
      {pills.map((pill) => {
```

**Review by olerichter00 — DISMISSED** [review]
Great work! 🌟

File: src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  const artworks = extractNodes(data.me?.alert?.artworksConnection)
+  const artworksCount = data.me?.alert?.artworksConnection?.counts?.total ?? 0
+  const artistIDs = data.me?.alert?.artistIDs ?? []
+  const areMoreMatchesAvailable =
+    artworksCount > NUMBER_OF_ARTWORKS_TO_SHOW && artistIDs.length === 1
+
+  const numWorks = `${artworksCount} ${pluralize("work", artworksCount)}`
+
+  const handleManageAlert = () => {
+    requestAnimationFrame(() => {
```
[the remark this answers is not in the document]
Comment (nickskalkin) [reply src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx]: I've just copied it from another similar screen. After testing it without requestAnimationFrame I can really tell a difference, so I'm removing it

File: src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+          <Text variant="sm">
+            {numWorks} currently on Artsy match your criteria. See our top picks for you:
+          </Text>
+          <Spacer y={1} />
+          <GenericGrid
+            width={screen.width - space(2)}
+            artworks={artworks}
+            hideSaveIcon
+            onPress={(slug: string) => {
+              requestAnimationFrame(() => {
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx:66]: I think we don't need this here

File: src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+  return (
+    <Flex>
+      {artworksCount > 1 && (
+        <Flex>
+          <Text variant="sm">
+            {numWorks} currently on Artsy match your criteria. See our top picks for you:
+          </Text>
+          <Spacer y={1} />
+          <GenericGrid
```
Comment (MounirDhahri) [inline src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx:61]: @gkartalis do we still need to keep this 👀

**Review by MounirDhahri — DISMISSED** [review]

File: src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx (hunk 4 above)
Comment (gkartalis) [reply src/app/Scenes/SavedSearchAlert/AlertArtworksGrid.tsx:61]: We do still use it in some places  searching for it it appears on 14 results in 12 files so I would say for now yes but in the future we could get rid of it

**Pull request merged**